Libplanet/Net/Swarm.cs

```diff
@@ -207,6 +206,13 @@ static Swarm()
         public event EventHandler<DifferentProtocolVersionEventArgs>
             DifferentVersionPeerEncountered;
 
+        /// <summary>
+        /// The <see cref="EventHandler" /> called when the block downloading is failed in
+        /// <see cref="PreloadAsync(
+        /// TimeSpan?, IProgress{PreloadState}, IImmutableSet{Address}, CancellationToken)" />
+        /// </summary>
+        public event EventHandler PreloadBlockDownloadFailed;
```

Comment:
Instead of `Swarm<T>` instances having this event, how about `Swarm<T>.PreloadAsync()` and `Swarm<T>.StartAsync()` methods taking a callback as an optional argument?

Reply:
Added `preloadBlockDownloadFailed` event handler to `Swarm<T>.PreloadAsync()` method and `Swarm<T>.StartAsync()` method as a parameter.